src/passes/Inlining.cpp

```diff
@@ -199,11 +235,55 @@ struct Inlining : public Pass {
     // return whether we did any work
     return inlined.size() > 0;
   }
+
+  bool canInline(int numUses) {
+    return numUses == SINGLE_USE;
+  }
+
+  bool worthInlining(Function* func) {
+    return Measurer::measure(func->body) <= INLINING_SIZE_LIMIT;
```

Comment:
Oh, yeah, we do want to limit re-optimization too. Anyway yeah fairly small is fine; I think inlining is something that really should be done at a higher level anyway. If it is done at the low level it should probably be on the basis of special knowledge of the target (i.e. wasm). So even this is kind of hacking around a limitation in LLVM (and even one that's solved by LTO, right?)